index.js

```diff
@@ -1241,21 +1257,28 @@
   //. use of the Either type to indicate completion (via a Right).
   //.
   //. ```javascript
-  //. > S.chainRec(Array,
-  //. .            s => s.length === 2 ? S.map(S.Right, [s + '!', s + '?'])
-  //. .                                : S.map(S.Left, [s + 'o', s + 'n']),
-  //. .            '')
+  //. > S.chainRec (Array)
+  //. .            (s => s.length === 2 ? S.map (S.Right) ([s + '!', s + '?'])
+  //. .                                 : S.map (S.Left) ([s + 'o', s + 'n']))
+  //. .            ('')
   //. ['oo!', 'oo?', 'on!', 'on?', 'no!', 'no?', 'nn!', 'nn?']
   //. ```
-  function chainRec(typeRep, f, x) {
-    function step(next, done, x) {
-      return Z.map(function(e) { return either(next, done, e); }, f(x));
-    }
-    return Z.chainRec(typeRep, step, x);
+  function chainRec(typeRep) {
+    return function(f) {
+      return function(x) {
+        return Z.chainRec (typeRep, step, x);
+      };
+      function step(next, done, x) {
+        return Z.map (either (next) (done), f (x));
+      }
```

Comment:
I decided to rely on function hoisting here to avoid interrupting the `function > return > > return` staircase.